### PR TITLE
Crate changes

### PR DESCRIPTION
**Pull request by FosterGrime-Heymynameslouie**
## About The Pull Request

Crates are now infinite in OP vendor

RO having to rely on crates from peoples orders, with their names on them, seems strange and this PR aims to remedy that.

~~To balance infinite crates, crate can **no longer be welded to receive 1 (one) metal.**~~

~~Now, Adds OP vendor specific crates that cannot be welded for 1 metal, designated as "mass produced"
Leaves pre placed crates and crates from req platform untouched.~~

Now, Now, adds autolathe specific printable crates that cannot be welded for 1 metal.

This PR is the sister PR of: #11950

thank you electric for helping me figure out how variables work

## Why It's Good For The Game

On hand crates for RO to smoothen out the customer service experience, having to rely less on crates from people's order is I believe a good qol change.

## Changelog
:cl:

add: infinite crates via op vendor in req
del: removed welding crates for 1 metal sheet on crates from op vendor
qol: made req supply chain easier

/:cl:

### CODE REVIEW

**Comment by Derrick2Electricboogaloo** [comment]
Yes, more of making it fucking annoying for engineers to scrap metal when shit hits the fan, how about we make it even simpler. Let crates be made from 1 or even 2 metal. And give ROs a small stack of metal to BUILD the damn things.

**Comment by maikilangiolo** [comment]
People absolutely malding they can't get one (1) metal from crates. Unreal

**Comment by Derrick2Electricboogaloo** [comment]
> People absolutely malding they can't get one (1) metal from crates. Unreal

Sometimes when shit hits the fan in an OP, you need those 100 crates worth of metal. It adds up really fast.

**Comment by Losenis** [comment]
> Yes, more of making it fucking annoying for engineers to scrap metal when shit hits the fan, how about we make it even simpler. Let crates be made from 1 or even 2 metal. And give ROs a small stack of metal to BUILD the damn things.

You need SEVEN (7) crates to make one cade with wire and an upgrade. If shit has gone down so hard that enough crates with req gear have been dropped to generate that much metal, marines threw so insanely hard that 2 or 3 cades are not going to cut it.

**Comment by Strayed-Raven** [comment]
just give req the ability to vend their own type of crates that cant be welded for metal while keeping crates that are ordered being able to be welded for metal.

**Comment by GreatKingWombat** [comment]
If crates don't give metal anymore for deconstructing, then there's no reason for them to cost welder fuel. Maybe make the req crates wooden instead, and make them deconstruct with a crowbar?

**Comment by FosterGrime-Heymynameslouie** [comment]
> Yes, more of making it fucking annoying for engineers to scrap metal when shit hits the fan, how about we make it even simpler. Let crates be made from 1 or even 2 metal. And give ROs a small stack of metal to BUILD the damn things.



> If crates don't give metal anymore for deconstructing, then there's no reason for them to cost welder fuel. Maybe make the req crates wooden instead, and make them deconstruct with a crowbar?



> just give req the ability to vend their own type of crates that cant be welded for metal while keeping crates that are ordered being able to be welded for metal.

new changes to pr address these concerns

**Review by Lumipharon — CHANGES_REQUESTED** [review]
I only have issue with the fact you can do funny things with infinite crates, like make massive areas difficult to pass through for xenos.

Are crates really in that short of supply that this is an issue? You can't sell them anymore, and you can use a labeller to name them.

**Comment by Lumipharon** [comment]
Integrity isn't the issue, it's the fact they're dense. If you fill an entire hall with 500 crates, it becomes a massive guh to clear them as xenos, even with 1 hit each.

**Comment by FosterGrime-Heymynameslouie** [comment]
> Integrity isn't the issue, it's the fact they're dense. If you fill an entire hall with 500 crates, it becomes a massive guh to clear them as xenos, even with 1 hit each.

made vendable crates non dense so anyone can pass through em now, hopefully negate any crate fob shenanigans

**Comment by FosterGrime-Heymynameslouie** [comment]
made it so that mass produced crates can be produced in autolathe instead, having to sacrifice metal materials. they cannot be welded for metal

File: code/game/objects/structures/crates_lockers/crates.dm [hunk 1]
```diff
@@ -331,3 +331,99 @@
...
+	if(!welder.isOn())
+		return FALSE
+
+	if(opened)
+		if(!welder.use_tool(src, user, 2 SECONDS, 1, 50))
+			to_chat(user, span_notice("You need more welding fuel to complete this task."))
+			return TRUE
+		visible_message(span_notice("\The [src] has been cut apart by [user] with [welder]."), "You hear welding.")
+		qdel(src)
+		return TRUE
```
Comment (Lumipharon) [inline code/game/objects/structures/crates_lockers/crates.dm:359]: This is not the way to do this, its just copy pasta code.
I'd recommend making the dropped material a var at the closet level, then you can just make the var null for this crate.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/crates_lockers/crates.dm (hunk 1 above)
Comment (FosterGrime-Heymynameslouie) [reply code/game/objects/structures/crates_lockers/crates.dm:355]: after alot of trial error I realized that I am too stupid to make this happen

can you help me figure out how to turn this into a variable?

**Comment by Losenis** [comment]
>If you fill an entire hall with 500 crates

If RO is spending the whole round sending down 500 empty crates one by one to some random asshole in the middle of nowhere instead of doing his job I'd sooner ~~go up and kill the motherfucker~~ ahelp it

**Comment by Losenis** [comment]
Also I'd argue crates should be dense purely because in a cluttered beacon you can walk into the beacon to push the sent down crate out of the way into an empty area to open, otherwise you're gonna spend 3 hours waiting for the alt-click to process the quadrillion items under that beacon. That's a far more common situation than the hypothetical crate river.

**Comment by Losenis** [comment]
>made it so that mass produced crates can be produced in autolathe instead, having to sacrifice metal materials. they cannot be welded for metal

What's the point now? Just order a single MRE and you get a free crate with it. The smallest amount you can order of metal is a WHOLE stack which you're not going to afford because if you can afford the points for reqstuff, you'd be ordering things for marines and get the crates from it. Not to mention that you'd also end up with the absurdly slow method of ordering metal, accepting order, lowering platform, bringing platform up, getting the metal, putting it in the autolathe, and THEN getting the crate to vend it. Compare to ordering a bunch of MREs in separate orders to bring up like 5 crates in 1 go.

This PR had promise but fearmongering is absolutely killing it.

**Comment by FosterGrime-Heymynameslouie** [comment]
made it so that drop materials are a var now

**Comment by FosterGrime-Heymynameslouie** [comment]
> > made it so that mass produced crates can be produced in autolathe instead, having to sacrifice metal materials. they cannot be welded for metal
> 
> What's the point now? Just order a single MRE and you get a free crate with it. The smallest amount you can order of metal is a WHOLE stack which you're not going to afford because if you can afford the points for reqstuff, you'd be ordering things for marines and get the crates from it. Not to mention that you'd also end up with the absurdly slow method of ordering metal, accepting order, lowering platform, bringing platform up, getting the metal, putting it in the autolathe, and THEN getting the crate to vend it. Compare to ordering a bunch of MREs in separate orders to bring up like 5 crates in 1 go.
> 
> This PR had promise but fearmongering is absolutely killing it.

you can make 4 crates from a single metal tile

**Comment by FosterGrime-Heymynameslouie** [comment]
> Also I'd argue crates should be dense purely because in a cluttered beacon you can walk into the beacon to push the sent down crate out of the way into an empty area to open, otherwise you're gonna spend 3 hours waiting for the alt-click to process the quadrillion items under that beacon. That's a far more common situation than the hypothetical crate river, which I've never seen happen even in 3h rounds where fob was absolutely packed to the brim with dropped crates that even the engis stopped giving a fuck about.

also forgot to mention, mass produced crates are dense now

File: code/game/objects/structures/crates_lockers/crates.dm [hunk 2]
```diff
@@ -331,3 +331,85 @@
...
+/obj/structure/closet/crate/mass_produced_crate
+	name = "Mass Produced Crate"
+	desc = "A rectangular steel crate. Cannot be welded for metal."
+	icon = 'icons/obj/structures/crates.dmi'
+	icon_state = "closed_basic"
+	icon_opened = "open_basic"
+	icon_closed = "closed_basic"
+	max_integrity = 5 //hopefully makes it so req crate spam can easily be destroyed
+	materials = list(/datum/material/metal = 200)
+	drop_material = FALSE
```
Comment (Lumipharon) [inline code/game/objects/structures/crates_lockers/crates.dm:344]: Minor thing but make this null, not FALSE.

Comment (FosterGrime-Heymynameslouie) [reply code/game/objects/structures/crates_lockers/crates.dm]: done and done

**Comment by Derrick2Electricboogaloo** [comment]
From how I read it, seems you changed it, so I change my vote! :)

**Comment by Losenis** [comment]
Funnily enough it's actually worse for engineers because ROs will keep the metal to themselves instead of sending it down groundside

**Pull request merged**